src/managedwriter/stream_connection.ts

```diff
@@ -107,20 +108,19 @@ export class StreamConnection extends EventEmitter {
   private handleError = (err: gax.GoogleError) => {
     this.trace('on error', err, JSON.stringify(err));
     if (this.shouldReconnect(err)) {
+      err.message = 'reconnect triggered due to: ' + err.message;
+      this.ackAllPendingWrites(err);
       this.reconnect();
```

Comment:
We no longer need this reconnect?

Reply:
Yeah, it will still be a very bad problem if we cannot distinguish connection level error vs request level error. No matter what we do, it could cause more duplications if we have request level error. We should put a comment here explaining the situation and revisit this (maybe one day the grpc layer will be fixed).

For now, let's remove gax.Status.RESOURCE_EXHAUSTED, from the retry code.

Reply:
I moved the logic to reconnect only when the connection is closed and there are pending writes and the check if an error is retryable, I removed the `gax.Status.RESOURCE_EXHAUSTED` status.